action.yml

```diff
@@ -0,0 +1,72 @@
+inputs:
+  fetch-targets:
+    description: 'depviz target to fetch (format: \"org1/repo1 org1/repo2 org2 repo1\")'
+    default: '' 
+  generation-targets:
+    description: 'depviz target to generate outputs (format: \"org1/repo1 org1/repo2 org2 repo1\")' 
+    default: $GITHUB_REPOSITORY
+  branch: 
+    description: 'the branch where to push the generated files.' 
+    default: 'depviz-generate'
+  json-output: 
+    description: 'i.e., docs/depviz.json; no generation if empty.'
+    default: ''
+  png-output: 
+    description: 'i.e., docs/depviz.png; no generation if empty.'
+    default: ''
+  svg-output:
+    description: 'i.e., docs/depviz.svg; no generation if empty.'
+    default: '' 
+  csv-output:
+    description: 'i.e., docs/depviz.csv; no generation if empty.'
+    default: ''
+  filters:
+    description: 'depviz filters use for the generation.'
+    required: false
+  depviz-version:
+    description: 'depviz version used to fetch and generate outputs.'
+    default: 'latest'
+  outputs-dirname:
+    description: 'name of the directory containing action outputs'
+    default: 'depviz-outputs'
+
+runs:
+  using: 'composite'
+  steps:
+  - uses: actions/setup-go@v4
+
+  # fetch
+  - run: go install moul.io/depviz/v3/cmd/depviz@${{ inputs.depviz-version }}
+    shell: bash
+    if: ${{ inputs.fetch-targets != '' }}
+  - run: depviz fetch --github-token ${GITHUB_TOKEN} ${{ inputs.fetch-targets }} 
+    shell: bash
+    if: ${{ inputs.fetch-targets != '' }}
+ 
+
+  # generate
+  - run: mkdir -p $(dirname ${{ inputs.outputs-dirname }}/${{ inputs.json-output }})
+    shell: bash
+
+  - run: depviz gen json ${{ inputs.generation-targets }} > ${{ inputs.outputs-dirname }}/${{ inputs.json-output }}
+    shell: bash
+    if: ${{ inputs.json-output != '' }}
+  - run: depviz gen graphviz ${{ inputs.generation-targets }} > ${{ inputs.outputs-dirname }}/${{ inputs.png-output }}
+    shell: bash
+    if: ${{ inputs.png-output != '' }}
+  - run: depviz gen graphviz ${{ inputs.generation-targets }} > ${{ inputs.outputs-dirname }}/${{ inputs.svg-output }}
+    shell: bash
+    if: ${{ inputs.svg-output != '' }}
+  - run: depviz gen svg ${{ inputs.generation-targets }} > ${{ inputs.outputs-dirname }}/${{ inputs.csv-output }}
+    shell: bash
+    if: ${{ inputs.csv-output != '' }}
+  
+  - uses: stefanzweifel/git-auto-commit-action@v4
+    env:          
+      GITHUB_TOKEN: $GITHUB_TOKEN
+    with:
+      file_pattern: ${{ inputs.outputs-dirname }}
+      commit_message: 'build: generate'
+      branch: ${{ inputs.branch }}
+      push_options: '--force'
+      create_branch: true
```

Comment:
```suggestion
```

This should be handled by the enduser, not by this action plugin. Let's move the snippet in the README.md as demo on how to do it.